perllib/FixMyStreet/DB/Result/Problem.pm

```diff
@@ -659,12 +659,23 @@ order of title.
 
 sub response_templates {
     my $problem = shift;
-    return $problem->result_source->schema->resultset('ResponseTemplate')->search(
+    my $schema = $problem->result_source->schema;
+    my @cids = $schema->resultset("Contact")->search(
         {
-            body_id => $problem->bodies_str_ids
+            body_id => $problem->bodies_str_ids,
+            category => $problem->category
+        }, {
+            columns => [ 'id' ],
+        }
+    )->get_column('id')->all;
+    return $schema->resultset('ResponseTemplate')->search(
+        {
+            body_id => $problem->bodies_str_ids,
+            contact_id => [ \@cids, undef ],
```

Comment:
Yes, I read it as selecting no categories was the same as not having made a filter, so applied to all, and would be a generic "applies anywhere" template. That seemed easier than having to select all the categories (a Select/Clear all would help there, yeah). I guess as you can't delete templates through the interface, no categories is a way to have it not shown, though; I think the best solution is probably some help text :) As either reading of nothing selected would be okay, depends what we want to do.